### PR TITLE
Ask for confirmation if kernel is busy

### PR DESCRIPTION
**Pull request by flying-sheep**
fixes ipython/ipython#8546

### CODE REVIEW

File: notebook/static/notebook/js/notebook.js [hunk 1]
```diff
@@ -249,6 +250,14 @@ define(function (require) {
             var cm_mode = langinfo.codemirror_mode || langinfo.name || 'null';
             that.set_codemirror_mode(cm_mode);
         });
+        
+        this.events.on('kernel_idle.Kernel', function () {
+            that.kernel_busy = false;
+        });
+        
+        this.events.on('kernel_busy.Kernel', function () {
+            that.kernel_busy = false;
```
Comment (takluyver) [inline notebook/static/notebook/js/notebook.js]: true, right?

File: notebook/static/notebook/js/notebook.js [hunk 2]
```diff
@@ -283,6 +292,10 @@ define(function (require) {
             if (kill_kernel) {
                 that.session.delete();
             }
+            // if the kernel is busy, prompt the user if he’s sure
+            if (that.kernel_busy) {
+                return "The Kernel is busy, changes may be lost.";
+            }
```
Comment (takluyver) [inline notebook/static/notebook/js/notebook.js]: I think this should be after the unsaved changes check - if there are unsaved changes and the kernel is busy, it's probably the unsaved changes that are more important.

I'd also tweak the wording of the message a bit: "... *outputs* may be lost".

File: notebook/static/notebook/js/notebook.js (hunk 1 above)
Comment (flying-sheep) [inline notebook/static/notebook/js/notebook.js]: lol of course :tired_face:

File: notebook/static/notebook/js/notebook.js (hunk 2 above)
Comment (flying-sheep) [inline notebook/static/notebook/js/notebook.js]: yeah… although firefox doesn’t recognize the message anyway.

**Comment by takluyver** [comment]
This looks fine to me. @minrk or @Carreau , can one of you take a look and merge if you're happy with it.

**Comment by Carreau** [comment]
I'm +1. @ellisonbg Might be picky on the phrasing.

**Pull request merged**